index.js

```diff
@@ -443,58 +445,93 @@ class HtmlWebpackInlineSVGPlugin {
 
     }
 
-
     /**
      * append the inlineImages SVG data to the output HTML and remove the original img
-     * @param {string} html
-     * @param {Object} inlineImage - parse5 document
+     * @param {Object{}} html
+     * @param {Object{}} inlineImage - parse5 document
+     * @param {Object{}} data - SVG data
+     * @param {Object{}} resolve - Consumer's Promise resolve hook
      * @returns {Promise}
      *
      */
-    processOutputHtml (html, inlineImage) {
+    optimizeTheSvgRetrievedData ({ html, inlineImage, data, resolve }) {
+        const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
 
-        return new Promise((resolve, reject) => {
+        const config = {}
 
-            const svgSrc = this.getImagesSrc(inlineImage)
 
+        // pass all objects to the config.plugins array
 
-            // if the image isn't valid resolve
+        config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
-            if (!svgSrc) return resolve(html)
 
+        // create a new instance of SVGO
+        // passing it the merged config, to optimize the svg
 
-            // read in the svg
+        const svgo = new SVGO(config)
 
-            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+        svgo.optimize(data)
+            .then((result) => {
 
-                if (err) reject(err)
+                const optimisedSVG = result.data
 
-                const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
+                html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
 
-                const config = {}
+                resolve(html)
 
+            })
+            .catch((err) => console.log(chalk.red(err)))
+    }
 
-                // pass all objects to the config.plugins array
 
-                config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
+    /**
+     * append the inlineImages SVG data to the output HTML and remove the original img by
+     * loading the SVG data from the filesystem or from the network
+     * @param {string} html
+     * @param {Object} inlineImage - parse5 document
+     * @returns {Promise}
+     *
+     */
+    processOutputHtmlReadingDataFromFileSystemOrNetwork (html, inlineImage) {
```

Comment:
I think the previous name was OK. Though I like your taste for descriptive names. 🙂
```suggestion
    this.processOutputHtml (html, inlineImage) {
```